pkg/services/ngalert/provisioning/accesscontrol.go

```diff
@@ -0,0 +1,53 @@
+package provisioning
+
+import (
+	"context"
+
+	ac "github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/auth/identity"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/store"
+)
+
+type RuleAccessControlService interface {
+	HasAccess(ctx context.Context, user identity.Requester, evaluator ac.Evaluator) (bool, error)
+	AuthorizeAccessToRuleGroup(ctx context.Context, user identity.Requester, rules models.RulesGroup) error
+	AuthorizeRuleChanges(ctx context.Context, user identity.Requester, change *store.GroupDelta) error
+}
+
+func newRuleAccessControlService(ac RuleAccessControlService) *provisioningRuleAccessControl {
+	return &provisioningRuleAccessControl{
+		RuleAccessControlService: ac,
+	}
+}
+
+type provisioningRuleAccessControl struct {
+	RuleAccessControlService
+}
+
+var _ ruleAccessControlService = &provisioningRuleAccessControl{}
+
+func (p *provisioningRuleAccessControl) AuthorizeAccessToRuleGroup(ctx context.Context, user identity.Requester, rules models.RulesGroup) error {
+	if can, err := p.CanReadAllRules(ctx, user); can || err != nil {
+		return err
+	}
+	return p.RuleAccessControlService.AuthorizeAccessToRuleGroup(ctx, user, rules)
+}
+
+func (p *provisioningRuleAccessControl) AuthorizeRuleChanges(ctx context.Context, user identity.Requester, change *store.GroupDelta) error {
```

Comment:
Massive nit -
These are a pair (they do the same thing for both read and write), can we align the naming closer? i.e. `AuthorizeRuleRead` and `AuthorizeRuleWrite`?

the term `Access` doesn't imply read, it implies write too. i think `CanReadAllRules`/`CanWriteAllRules` does this very well already